### PR TITLE
doc: add API compatibility section for golang filter.

### PR DESCRIPTION
**Pull request by doujiang24**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/configuration/http/http_filters/golang_filter.rst [hunk 1]
```diff
@@ -45,6 +45,19 @@ For example, to build the ``.so`` for a ``foo`` plugin, you might run:
 
    $ bazel run @go_sdk//:bin/go build -- --buildmode=c-shared  -v -o path/to/output/libfoo.so path/to/src/foo
 
+API compatibility
```
Comment (wangfakang) [inline docs/root/configuration/http/http_filters/golang_filter.rst:48]: Would it be better to move this to the 'Attention' section of `Developing a Go plugin`.

File: docs/root/configuration/http/http_filters/golang_filter.rst [hunk 2]
```diff
@@ -45,6 +45,19 @@ For example, to build the ``.so`` for a ``foo`` plugin, you might run:
...
+~~~~~~~~~~~~~~~~~
+
+.. attention::
+  The Go plugin API is not yet stable, you are **strong** recommended to use the same version of Go plugin SDK and Envoy.
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/golang_filter.rst:52]:
```suggestion
  The Go plugin API is not yet stable, you are **strongly** recommended to use the same version of Go plugin SDK and Envoy.
```

File: docs/root/configuration/http/http_filters/golang_filter.rst [hunk 3]
```diff
@@ -45,6 +45,19 @@ For example, to build the ``.so`` for a ``foo`` plugin, you might run:
...
+
+When you are using a release version of Envoy, i.e. 1.26.x,
+you should use `github.com/envoyproxy/envoy v1.26.x` in the go.mod file.
+
+When you are not using a release, i.e. the latest main branch of Envoy,
+you could use `go get -u github.com/envoyproxy/envoy@SHA` to get the same version of Go plugin SDK,
+the `SHA` is the latest commit sha.
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/golang_filter.rst:59]:
```suggestion
the SHA is the latest commit sha.
```

File: docs/root/configuration/http/http_filters/golang_filter.rst [hunk 4]
```diff
@@ -45,6 +45,19 @@ For example, to build the ``.so`` for a ``foo`` plugin, you might run:
...
+~~~~~~~~~~~~~~~~~
+
+.. attention::
+  The Go plugin API is not yet stable, you are **strong** recommended to use the same version of Go plugin SDK and Envoy.
+
+When you are using a release version of Envoy, i.e. 1.26.x,
+you should use `github.com/envoyproxy/envoy v1.26.x` in the go.mod file.
+
+When you are not using a release, i.e. the latest main branch of Envoy,
+you could use `go get -u github.com/envoyproxy/envoy@SHA` to get the same version of Go plugin SDK,
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/golang_filter.rst:58]:
```suggestion
you could use ``go get -u github.com/envoyproxy/envoy@SHA`` to get the same version of Go plugin SDK,
```

**Review by phlax — COMMENTED** [review]
thanks for working on this @doujiang24 - added a few formatting nits

**Comment by doujiang24** [comment]
@wangfakang @phlax Thanks! all addressed.

**Review by wangfakang — APPROVED** [review]
Thanks. LGTM

**Review by phlax — APPROVED** [review]
lgtm, thanks @doujiang24

**Pull request merged**